query/src/sql/planner/semantic/type_check.rs

```diff
@@ -1206,39 +1192,13 @@ impl<'a> TypeChecker<'a> {
                 };
                 Some(self.resolve_function(span, "version", &[&arg], None).await)
             }
-            "user" => match self.ctx.get_current_user() {
-                Ok(user) => {
-                    let arg = Expr::Literal {
-                        span: &[],
-                        lit: Literal::String(user.identity().to_string()),
-                    };
-                    Some(self.resolve_function(span, "user", &[&arg], None).await)
-                }
-                Err(e) => Some(Err(e)),
-            },
-            "currentuser" => match self.ctx.get_current_user() {
-                Ok(user) => {
-                    let arg = Expr::Literal {
-                        span: &[],
-                        lit: Literal::String(user.identity().to_string()),
-                    };
-                    Some(
-                        self.resolve_function(span, "currentuser", &[&arg], None)
-                            .await,
-                    )
-                }
-                Err(e) => Some(Err(e)),
-            },
-            "current_user" => match self.ctx.get_current_user() {
+            "user" | "currentuser" | "current_user" => match self.ctx.get_current_user() {
```

Comment:
`name @("user" | "currentuser" | "current_user") => `

may be better, so we have a `name` variable.